### DISCUSSION TITLE
`LoadAsync` Interceptor.OnLoad() is not called

### DISCUSSION THREAD

**Discussion by cd21h** [nhibernate/nhibernate-core#2602] [Q&A]
Hello guys,
I implemented interceptor `OnLoad` and it works with `GetAsync` but not called if entity is loaded using `LoadAsync`. 
Is this by design?

**Comment by bahusoid** [comment]
`session.LoadAsync` call doesn't trigger actual entity loading it just returns proxy (if lazy loading is supported). So yes not calling `OnLoad` (which is about actual entity loading from DB) here is by design. Or are you saying it's not called even when lazy entity is actually initialized?

**Reply by cd21h** [reply]
Sorry I had to be more specific.
`
var x = await _session.LoadAsync<Entity>(1, cancellationToken).ConfigureAwait(false);
var y = await _session.GetAsync<Entity>(1, cancellationToken).ConfigureAwait(false);
`

I do not observe interceptor is called.

Also interceptor is injected during the session creation 
` _session.WithOptions().Interceptor(x).OpenSession() `
